File: src/Calibration/Pose.cpp
#include <ZividPython/Calibration/Pose.h>
#include <ZividPython/Matrix.h>

#include <pybind11/pybind11.h>

namespace py = pybind11;

namespace ZividPython
{
    void wrapClass(pybind11::class_<Zivid::Calibration::Pose> pyClass)
    {
        pyClass
            .def(py::init([](const Eigen::Matrix<float, 4, 4, Eigen::RowMajor> &matrix) {
                return std::make_unique<Zivid::Calibration::Pose>(Conversion::toCpp(matrix));
            }))
            .def("to_matrix", [](const Zivid::Calibration::Pose &pose) { return Conversion::toPy(pose.toMatrix()); });
    }
} // namespace ZividPython

File: modules/zivid/_calibration/_pose.py
import _zivid
import numpy as np


class Pose:
    def __init__(self, transformation_matrix):
        self.__impl = _zivid.calibration.Pose(transformation_matrix)

    def to_matrix(self):
        return self.__impl.to_matrix()

    def __str__(self):
        return str(self.__impl)
